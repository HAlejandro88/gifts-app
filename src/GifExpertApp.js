import React,{ useState } from 'react';
import AddCategory from './components/AddCategory';
import { GifGrid } from './components/GifGrid';


const GifExpertApp = ({ defaultCategorie = [] }) => {
    
    const [categories, setCategories] = useState(defaultCategorie);

    /* const [categories, setCategories] = useState(['Dragon Ball']); */


    return (
        <>
            <h2>GifexpertApp</h2>
            <AddCategory setCategories={ setCategories } />
            <hr/>

            <ol>
                {
                    categories.map(categoria => <GifGrid key={ categoria } category={ categoria } />)
                }
            </ol>
        </>
    )
}

export default GifExpertApp
