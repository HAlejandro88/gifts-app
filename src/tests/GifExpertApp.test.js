import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas en la aplicacion de <GifExpertApp />', () => {
    test('comparar contra el snapshot', () => {
        const wrapper = shallow(<GifExpertApp />)
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar una lista de categorias', () => {
        const categorys = ['dragon ball', 'batman'];
        const wrapper = shallow(<GifExpertApp defaultCategorie={ categorys } />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categorys.length);
    });
});