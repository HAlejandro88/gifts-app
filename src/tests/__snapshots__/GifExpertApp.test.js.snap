// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en la aplicacion de <GifExpertApp /> comparar contra el snapshot 1`] = `
<Fragment>
  <h2>
    GifexpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en la aplicacion de <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifexpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="dragon ball"
      key="dragon ball"
    />
    <GifGrid
      category="batman"
      key="batman"
    />
  </ol>
</Fragment>
`;
